custom_components/mikrotik_router/mikrotik_controller.py

```diff
@@ -2015,7 +2014,128 @@ def _address_part_of_local_network(self, address):
     #   _get_accounting_uid_by_ip
     # ---------------------------
     def _get_accounting_uid_by_ip(self, requested_ip):
-        for mac, vals in self.data["accounting"].items():
+        for mac, vals in self.data["client_traffic"].items():
+            if vals.get("address") is requested_ip:
+                return mac
+        return None
+
+    # ---------------------------
+    #   _get_iface_from_entry
+    # ---------------------------
+    def _get_iface_from_entry(self, entry):
+        """Get interface default-name using name from interface dict"""
+        uid = None
+        for ifacename in self.data["interface"]:
+            if self.data["interface"][ifacename]["name"] == entry["interface"]:
+                uid = ifacename
+                break
+
+        return uid
+
+    # ---------------------------
+    #   process_kid_control
+    # ---------------------------
+    def process_kid_control_devices(self):
+        """Get Kid Control Device data from Mikrotik"""
+
+        uom_type, uom_div = self._get_unit_of_measurement()
+
+        # Build missing hosts from main hosts dict
+        for uid, vals in self.data["host"].items():
+            if uid not in self.data["client_traffic"]:
+                self.data["client_traffic"][uid] = {
+                    "address": vals["address"],
+                    "mac-address": vals["mac-address"],
+                    "host-name": vals["host-name"],
+                    "previous-bytes-up": 0.0,
+                    "previous-bytes-down": 0.0,
+                    "wan-tx": 0.0,
+                    "wan-rx": 0.0,
+                    "tx-rx-attr": uom_type,
+                    "available": False,
+                    "local_accounting": False,
+                }
+
+        _LOGGER.debug(f"Working with {len(self.data['client_traffic'])} kid control devices")
+
+        time_diff = self.api.take_client_traffic_snapshot(False)
+        if not time_diff:
+            return
+
+        kid_control_devices_data = parse_api(
```

Comment:
Can we merge this into get_kidcontrol so its run all in one query? This may be causing lot of queries in large networks.

Reply:
Hmm, I'm not really sure, KidControl feature is a separate feature that requires another checkbox through config flow, thats why i didn't to use it, so I don't mix two features into one. This seemed like a better approach, let user config KidControl through Mikrotik to get KidControlDevices and then select through integration which fetaure will be polled.

About lot of queries, I'm not really sure why, this is only one api call to /ip/kid-control/device which returns multiple rows - one for each device, I have about 50 devices ATM and everything works fine.

Reply:
oh right, I misread something there